core/src/main/scala/org/apache/spark/scheduler/cluster/CoarseGrainedSchedulerBackend.scala

```diff
@@ -272,11 +272,16 @@ class CoarseGrainedSchedulerBackend(scheduler: TaskSchedulerImpl, val rpcEnv: Rp
         removeWorker(workerId, host, message)
         context.reply(true)
 
-      case RetrieveSparkAppConfig =>
+      case RetrieveSparkAppConfig(resourceProfileId) =>
+        // note this will be updated in later prs to get the ResourceProfile from a
+        // ResourceProfileManager that matches the resource profile id
+        // for now just use default profile
+        val rp = ResourceProfile.getOrCreateDefaultProfile(conf)
         val reply = SparkAppConfig(
           sparkProperties,
           SparkEnv.get.securityManager.getIOEncryptionKey(),
-          Option(delegationTokens.get()))
+          Option(delegationTokens.get()),
+          rp)
```

Comment:
Executors do not know what default ResourceProfile is? If default ResourceProfile is derived from Spark configs, do we still need to send a default ResourceProfile to executors? Can executors generate it if it knows it uses default one?

Reply:
They can but its really just extra code and an extra code path to have to maintain.   I specifically removed the old code that was generating the default profile on the executor side from the configs because the driver already calculated it and there isn't a reason to have 2 paths to calculate it.  This way it all works the same. Was there something specific here you are concerned with?

Reply:
Oh I see. I had thought that if we don't have special resource need, we don't need such ResourceProfile communication between driver/executors. If for a clean code path and the communication is cheap, I think it should be fine.